bake/bake_test.go

```diff
@@ -1363,3 +1363,110 @@ func TestJSONNullVars(t *testing.T) {
+
+func TestAttestOverride(t *testing.T) {
+	ctx := context.TODO()
+
+	// file without attest set
+	fp := File{
+		Name: "docker-bake.hcl",
+		Data: []byte(`target "default" {}`),
+	}
+
+	// no override
+	m, _, err := ReadTargets(ctx, []File{fp}, []string{"default"}, nil, nil)
+	require.NoError(t, err)
+	require.Empty(t, m["default"].Attest)
+
+	opts, err := TargetsToBuildOpt(m, &Input{})
+	require.NoError(t, err)
+	require.Empty(t, opts["default"].Attests)
+
+	// with override
+	m, _, err = ReadTargets(ctx, []File{fp}, []string{"default"}, []string{"*.attest=type=sbom,generator=custom"}, nil)
+	require.NoError(t, err)
+	require.Equal(t, []string{"type=sbom,generator=custom"}, m["default"].Attest)
+
+	opts, err = TargetsToBuildOpt(m, &Input{})
+	require.NoError(t, err)
+	require.Equal(t, map[string]*string{"sbom": ptrstr("type=sbom,generator=custom")}, opts["default"].Attests)
+
+	// with disabled=true override
+	m, _, err = ReadTargets(ctx, []File{fp}, []string{"default"}, []string{"*.attest=type=sbom,disabled=true"}, nil)
+	require.NoError(t, err)
+	require.Equal(t, []string{"type=sbom,disabled=true"}, m["default"].Attest)
+
+	opts, err = TargetsToBuildOpt(m, &Input{})
+	require.NoError(t, err)
+	require.Equal(t, map[string]*string{"sbom": nil}, opts["default"].Attests)
+
+	// with disabled=false override
+	m, _, err = ReadTargets(ctx, []File{fp}, []string{"default"}, []string{"*.attest=type=sbom,disabled=false"}, nil)
+	require.NoError(t, err)
+	require.Equal(t, []string{"type=sbom,disabled=false"}, m["default"].Attest)
+
+	opts, err = TargetsToBuildOpt(m, &Input{})
+	require.NoError(t, err)
+	require.Equal(t, map[string]*string{"sbom": ptrstr("type=sbom,disabled=false")}, opts["default"].Attests)
+
+	// file with attest set
+	fp = File{
+		Name: "docker-bake.hcl",
+		Data: []byte(
+			`target "default" {
+				attest = ["type=sbom,generator=custom"]
+			}`),
+	}
+
+	// no override
+	m, _, err = ReadTargets(ctx, []File{fp}, []string{"default"}, nil, nil)
+	require.NoError(t, err)
+	require.Equal(t, []string{"type=sbom,generator=custom"}, m["default"].Attest)
+
+	opts, err = TargetsToBuildOpt(m, &Input{})
+	require.NoError(t, err)
+	require.Equal(t, map[string]*string{"sbom": ptrstr("type=sbom,generator=custom")}, opts["default"].Attests)
+
+	// with duplicate override
+	m, _, err = ReadTargets(ctx, []File{fp}, []string{"default"}, []string{"*.attest=type=sbom,generator=custom"}, nil)
+	require.NoError(t, err)
+	require.Equal(t, []string{"type=sbom,generator=custom"}, m["default"].Attest)
+
+	opts, err = TargetsToBuildOpt(m, &Input{})
+	require.NoError(t, err)
+	require.Equal(t, map[string]*string{"sbom": ptrstr("type=sbom,generator=custom")}, opts["default"].Attests)
+
+	// with disabled=true override
+	m, _, err = ReadTargets(ctx, []File{fp}, []string{"default"}, []string{"*.attest=type=sbom,disabled=true"}, nil)
+	require.NoError(t, err)
+	require.Equal(t, []string{"type=sbom,generator=custom", "type=sbom,disabled=true"}, m["default"].Attest)
```

Comment:
I don't really understand how these make sense. If the user sets that sbom should be disabled then why don't we just set it like that.

I think `--set` should either replace the attestation with the same type or append the non-type keys. Adding a second record doesn't make sense to me. If that sounds too magical then we can replace the whole block.